### PR TITLE
Task-52830 : Add documentation to explain how to configure password complexity

### PR DESCRIPTION
**Pull request by rdenarie**

### CODE REVIEW

File: docs/Configuration.rst [hunk 1]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
 
+**Configuration of password validator**
+
+By default, the password will be validated as follows:
+
+-  The length must be between 9 and 256 characters.
+
+-  Must contains at minima 1 digit, 1 lower case, 1 upper case
+
+This default configuration correspond to the minimal password length required by
```
Comment (plamarque) [inline docs/Configuration.rst]:
```suggestion
This default configuration correspond to the minimal password length recommended by
```

File: docs/Configuration.rst [hunk 2]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
+`ANSSI <https://www.ssi.gouv.fr/uploads/2021/10/anssi-guide-authentification_multifacteur_et_mots_de_passe.pdf>`__
```
Comment (plamarque) [inline docs/Configuration.rst]:
```suggestion
`ANSSI-PG-078 <https://www.ssi.gouv.fr/uploads/2021/10/anssi-guide-authentification_multifacteur_et_mots_de_passe.pdf>`__ in low to medium sensitivity contexts.
```

File: docs/Configuration.rst [hunk 3]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
+
+If you want to change the password validation to use the middle level defined by ANSSI, you could use the following
```
Comment (plamarque) [inline docs/Configuration.rst]:
```suggestion
If you want to change the password validation in medium to high sensitivity contexts, the ANSSI recommends to use the following the following
```

File: docs/Configuration.rst [hunk 4]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
+
+::
+
+    # validators
+    gatein.validators.passwordpolicy.regexp=((?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).\\{12,256\\})
+    gatein.validators.passwordpolicy.length.max=256
+    gatein.validators.passwordpolicy.length.min=12
+    gatein.validators.passwordpolicy.format.message=Minimum of 1 digit, 1 lower case, 1 upper case, minimum of 12 chars.
+
+If you want to change the password validation to use the strong level defined by ANSSI, you could use the following configuration :
```
Comment (plamarque) [inline docs/Configuration.rst:2129]:
```suggestion
If you want to change the password validation in high to very high sensitivity contexts, the ANSSI recommends to use the following the following configuration
```

File: docs/Configuration.rst [hunk 5]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
+::
+
+    # validators
+    gatein.validators.passwordpolicy.regexp=((?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).\\{15,256\\})
+    gatein.validators.passwordpolicy.length.max=256
+    gatein.validators.passwordpolicy.length.min=15
+    gatein.validators.passwordpolicy.format.message=Minimum of 1 digit, 1 lower case, 1 upper case, minimum of 15 chars.
+
+
+.. note:: Even if we define a maximum for the password length, we do not display it to the user. It a technical limit to prevent
```
Comment (plamarque) [inline docs/Configuration.rst]:
```suggestion
.. note:: Even when the maximum password length is not set, that maximum will not be revealed to the end user. This is done on purpose to limit exposure to
```

File: docs/Configuration.rst [hunk 6]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
+          some attacks related to very long password.
+
+.. note:: If you need to define more rules for password complexity, you can update the regexp according to your rules. Remember
+          to update the error message according to your modifications.
```
Comment (plamarque) [inline docs/Configuration.rst]:
```suggestion
          to update the error message accordingly to help the user enter a valid password.
```

**Review by plamarque — CHANGES_REQUESTED** [review]
reviewed, see my suggestions.

File: docs/Configuration.rst [hunk 7]
```diff
@@ -2101,6 +2104,46 @@ warning message:
...
+By default, the password will be validated as follows:
+
+-  The length must be between 9 and 256 characters.
+
+-  Must contains at minima 1 digit, 1 lower case, 1 upper case
+
+This default configuration correspond to the minimal password length recommended by
+`ANSSI-PG-078 <https://www.ssi.gouv.fr/uploads/2021/10/anssi-guide-authentification_multifacteur_et_mots_de_passe.pdf>`__ in low to medium sensitivity contexts.
+
+If you want to change the password validation to use the middle level defined by ANSSI, you could use the following
```
Comment (rdenarie) [inline docs/Configuration.rst]:
```suggestion
If you want to change the password validation in medium to high sensitivity contexts, the ANSSI recommends using the following
```

**Review by plamarque — APPROVED** [review]
LGTM

**Pull request merged**